src/main/java/org/terasology/dynamicCities/parcels/DynParcelTypeHandler.java

```diff
@@ -15,41 +15,41 @@
  */
 package org.terasology.dynamicCities.parcels;
 
-
 import com.google.common.collect.ImmutableMap;
 import org.terasology.commonworld.Orientation;
 import org.terasology.math.geom.Rect2i;
-import org.terasology.persistence.typeHandling.DeserializationContext;
 import org.terasology.persistence.typeHandling.PersistedData;
 import org.terasology.persistence.typeHandling.PersistedDataMap;
 import org.terasology.persistence.typeHandling.RegisterTypeHandler;
-import org.terasology.persistence.typeHandling.SerializationContext;
-import org.terasology.persistence.typeHandling.SimpleTypeHandler;
+import org.terasology.persistence.typeHandling.*;
 
 import java.util.Map;
+import java.util.Optional;
 
 @RegisterTypeHandler
-public class DynParcelTypeHandler extends SimpleTypeHandler<DynParcel> {
+public class DynParcelTypeHandler extends TypeHandler<DynParcel> {
 
     @Override
-    public PersistedData serialize(DynParcel parcel, SerializationContext context) {
+    public PersistedData serializeNonNull(DynParcel parcel, PersistedDataSerializer serializer) {
         Map<String, PersistedData> data = new ImmutableMap.Builder()
-                .put("height", context.create(parcel.getHeight()))
-                .put("posX", context.create(parcel.getShape().minX()))
-                .put("posY", context.create(parcel.getShape().minY()))
-                .put("sizeX", context.create(parcel.getShape().sizeX()))
-                .put("sizeY", context.create(parcel.getShape().sizeY()))
-                .put("zone", context.create(parcel.getZone()))
-                .put("orientation", context.create(parcel.getOrientation().name()))
+                .put("height", serializer.serialize(parcel.getHeight()))
+                .put("posX", serializer.serialize(parcel.getShape().minX()))
+                .put("posY", serializer.serialize(parcel.getShape().minY()))
+                .put("sizeX", serializer.serialize(parcel.getShape().sizeX()))
+                .put("sizeY", serializer.serialize(parcel.getShape().sizeY()))
+                .put("zone", serializer.serialize(parcel.getZone()))
+                .put("orientation", serializer.serialize(parcel.getOrientation().name()))
                 .build();
-        return context.create(data);
+        return serializer.serialize(data);
     }
 
     @Override
-    public DynParcel deserialize(PersistedData data, DeserializationContext context) {
+    public Optional<DynParcel> deserialize(PersistedData data) {
         PersistedDataMap root = data.getAsValueMap();
```

Comment:
It wasn't part of the old code, but I think it'll be good to return `Optional.empty()` when `data` is not a value map:
```java
if (!data.isValueMap()) {
    return Optional.empty();
}
```
If `data` is not a value map, the input is in an invalid format and the TypeHandler API convention in this case is to return `Optional.empty()`. If this check is not done, then `getAsValueMap` will throw an exception when `data` is not a value map, which we want to avoid.

Reply:
@PutawanDE In my opinion, this `if` statement should be the first piece of code in the method. Right now we are getting the data as a value map and checking it afterwards, you should invert this order.

Reply:
@PutawanDE In my opinion, this if statement should be the first piece of code in the method. Right now we are getting the data as a value map and checking it afterwards, you should invert this order.